app/src/main/java/com/myperishableplanner/v21001/ItemViewModel.kt

```diff
@@ -40,9 +43,9 @@ class ItemViewModel (var itemService: IItemService = ItemService()): ViewModel()
             firestore.collection("itemDetail").document(itemDetail.itemDetailId)
         }
         itemDetail.itemDetailId = document.id
-        document.set (itemDetail)
-        val handle = document.set (itemDetail)
+
+        val handle = document.set(itemDetail)
         handle.addOnSuccessListener { Log.d("Firebase","Document Saved")}
-        handle.addOnFailureListener { Log.e("Firebase","Document Saved")}
+        handle.addOnFailureListener { Log.e("Firebase","Failed to save document")}
```

Comment:
Changed the log message for the on failure callback to indicate a failure to save document.

Reply:
Good.  Add the exception ($it) to the message as well, so the developer understands why it failed.